### PR TITLE
feat(elements): add logto-username element

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

This PR introduces the `logto-username` component and its supporting infrastructure for the Logto Elements package. Key changes include:

- Added new components:
  - LogtoIcon: A custom element for consistent icon rendering
  - LogtoProfileItem: A custom element for displaying profile information
  - LogtoUsername: A component to display the user's username
- Created a base class LogtoProfileItemElement to handle common functionality for profile item components
  - Implemented unit tests for the LogtoUsername component
- Added necessary icons and SVG type declarations
- Updated the main index.html file to include the new LogtoUsername component and required setup
- Added a new CSS file (style.css) with root variables for consistent styling
- Updated index.ts and react.ts to export the new components

## Testing

Test locally and UT.

image

## Checklist

- [ ] `.changeset`
- [x] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/elements/src/account/components/logto-profile-item.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  `;
+
+  render() {
+    return html`
+      <div class="label">
+        <slot name="label-icon"></slot>
+        <slot name="label-text"></slot>
+      </div>
+      <slot name="content"><span class="no-value">Not set</span></slot>
+      <slot name="actions"></slot>
```
Comment (simeng-li) [inline packages/elements/src/account/components/logto-profile-item.ts:84]: Just curious will there be an extra wrapper element in dom when passing in a child element to the slot?

File: packages/elements/src/account/elements/logto-username.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { LogtoProfileItemElement } from './LogtoProfileItemElement.js';
+
+const tagName = 'logto-username';
+
+@customElement(tagName)
+export class LogtoUsername extends LogtoProfileItemElement {
+  static tagName = tagName;
+
+  protected isAccessible() {
+    return this.accountContext?.userProfile.username !== undefined;
```
Comment (simeng-li) [inline packages/elements/src/account/elements/logto-username.ts:16]: should we display a placeholder here if the username is empty but editable?

File: packages/elements/src/account/elements/logto-username.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { when } from 'lit/directives/when.js';
+
+import usernameIcon from '../icons/username.svg';
+
+import { LogtoProfileItemElement } from './LogtoProfileItemElement.js';
+
+const tagName = 'logto-username';
+
+@customElement(tagName)
+export class LogtoUsername extends LogtoProfileItemElement {
```
Comment (simeng-li) [inline packages/elements/src/account/elements/logto-username.ts:12]: Add some jsDoc comments here? Briefly describe the usage of this class and maybe key properties?

File: packages/elements/src/account/elements/logto-username.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/elements/src/account/elements/logto-username.ts:16]: We have, the `Not set` will be rendered when a username is not set.
image

File: packages/elements/src/account/elements/logto-username.ts (hunk 3 above)
Comment (xiaoyijun) [reply packages/elements/src/account/elements/logto-username.ts:12]: Will add all comments with other elements later.

File: packages/elements/src/account/components/logto-profile-item.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/elements/src/account/components/logto-profile-item.ts:84]: The whole slot will be replaced with provided element.

**Review by wangsijie — APPROVED** [review]

**Pull request merged**